### ISSUE TITLE
[Train] Allow passing in placement group to trainer  

### ISSUE THREAD

**Issue by jiaodong** [ray-project/ray#28924]
### Description

From community user:

```python
trainer = TorchTrainer(
    train_func,
    train_loop_config = {
      "lr": args.learning_rate, 
      "batch_size": args.batch_size, 
      "epochs": args.epochs
    },
    scaling_config = ScalingConfig(
      num_workers = args.num_workers, 
      use_gpu = args.use_gpu,
      resources_per_worker = {"CPU": args.cpus_per_worker},
      trainer_resources = {"my_custom_resources": 1}
    )
  )
```

"
Here I can specify the number of workers, the number of CPUs per worker and even some custom resources if I need them, but the placement group is automatically created by ray somewhere during the execution of trainer.fit(). What I was asking is if there's a way to create the placement group in advance and tell the trainer to run on that placement group specifically.

I am running some training jobs to measure their performance (execution times, etc.). However, it takes a while to spin-up nodes in the cluster, that's why I was hoping to be able to reserve resources in advance 
"

**Comment by cszhu** [comment]
This P2 issue has seen no activity in the past 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please comment and remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
